iptables/utils_for_test.go

```diff
@@ -330,17 +350,43 @@ func (d *restoreCmd) Run() error {
 			d.Dataplane.ChainMods.Add(chainMod{name: chainName, ruleNum: ruleNum})
 		case "-D", "--delete":
 			chainName = parts[1]
-			Expect(len(parts)).To(Equal(3), "--delete only expects two arguments")
-			ruleNum, err := strconv.Atoi(parts[2]) // 1-indexed position of rule.
-			Expect(err).NotTo(HaveOccurred())
-			ruleIdx := ruleNum - 1 // 0-indexed array index of rule.
+			// Deleting cali chains by index
+			if strings.HasPrefix(chainName, "cali") {
```

Comment:
This is good enough but it'd be clearer if this code was structured to parser the rule and then assert on it, rather than having special cases.  I.e.
```
if <second arg is numeric> {
    <assert no more args>
    <assert this is a cali rule, with message like "delete-by-index used for non-calico chain, could cause races">
    <handle the deletion>
} else {
    <handle the deletion>
}
```

Reply:
That's much better; updated.